src/lib/statistics/statistics_objects/range_filter.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include "abstract_statistics_object.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+//! default number of maximum range count
+static constexpr uint32_t MAX_RANGES_COUNT = 10;
+
+/**
+ * Filter that stores a certain number of value ranges. Each range represents a spread
+ * of values that is contained within the bounds.
+ * Example: [1, 2, 4, 7] might be represented as [1, 7]
```

Comment:
This example would also serve for the MinMaxFilter, wouldn't it? Can we find a better one?